### PR TITLE
Add statistics display to beatmap card

### PR DESCRIPTION
**Pull request by bdach**
More pieces of #14216.

2021-11-07-210923_541x231_scrot

I've decided to make each statistic item an individual component for two reasons: composability (for ease of implementation of the other sizes in some unspecified future), and because they end up actually having some degree of logic in them. Testing is mostly visual via the test scene, I'll cover properly using assertions on request.

I probably haven't split *all* of the required logic per se but I'll leave further code reuse until that is actually needed, i.e. until  when the other card sizes are going to be implemented. Unsure if I'm going to be doing that.

References for new added `APIBeatmapSet` fields:

* [`hype`](https://github.com/ppy/osu-web/blob/93930cd02cfbd49724929912597c727c9fbadcd1/app/Transformers/BeatmapsetCompactTransformer.php#L52-L55)
* [`nominations_summary`](https://github.com/ppy/osu-web/blob/8ed7d071fd1d3eaa7e43cf0e4ff55ca2fef9c07c/app/Models/Beatmapset.php#L1009-L1015) (referenced [here](https://github.com/ppy/osu-web/blob/8ed7d071fd1d3eaa7e43cf0e4ff55ca2fef9c07c/app/Transformers/BeatmapsetTransformer.php#L37))

### CODE REVIEW

File: osu.Game/Beatmaps/Drawables/Cards/Statistics/BeatmapCardDateStatistic.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            this.dateTime = dateTime;
+
+            Icon = FontAwesome.Regular.CheckCircle;
+            Text = dateTime.ToLocalisableString(@"d MMM yyyy");
+        }
+
+        public override object TooltipContent => dateTime;
+        public override ITooltip GetCustomTooltip() => new DateTooltip();
+
+        public static BeatmapCardDateStatistic? CreateFor(IBeatmapSetOnlineInfo beatmapSetInfo)
```
Comment (peppy) [inline osu.Game/Beatmaps/Drawables/Cards/Statistics/BeatmapCardDateStatistic.cs:29]: Hmm, I can see why this was required, but not sure about the structure. I guess the reasoning for putting it here is that this component is going to be reused in other places and placing the `displayDateFor` code in `BeatmapCard` would disallow easy reuse?

**Review by peppy — COMMENTED** [review]
Looks fine apart from the one discussion point.

File: osu.Game/Beatmaps/Drawables/Cards/Statistics/BeatmapCardDateStatistic.cs (hunk 1 above)
Comment (bdach) [reply osu.Game/Beatmaps/Drawables/Cards/Statistics/BeatmapCardDateStatistic.cs:29]: Right. If the appropriate date is null in the API response I somehow need to not show anything at all. There are many ways to do this, most of which are shaped something like this, though.

I could replace `null` with `Empty()`, or I could try setting alpha to 0 in such cases - the second option may be a bit WTF if someone happens to need to debug it, though.

Comment (peppy) [reply osu.Game/Beatmaps/Drawables/Cards/Statistics/BeatmapCardDateStatistic.cs:29]: I think it's alright, just a different pattern to what we've done until now. We'll see if it sticks.

**Review by peppy — APPROVED** [review]

**Pull request merged**